### ISSUE TITLE
No-effects opengl test render

### ISSUE THREAD

**Issue by age3rcm** [opentoonz/opentoonz#4563]
### Is your feature request related to a problem? Please describe

Big rendering times when FX are applied or anti-aliased when you just want to have a movie file with the core animation

### Describe the solution you'd like

https://www.youtube.com/watch?v=j7nx3XaXq_Y
Like in the example, show a possibility to export the animation without the visual effects
it will be a quicker export to be able to visualize how the animation is layered out both to the animator and possible client

**Comment by DarrenTAnims** [comment]
You can turn off the effects by either unplugging them or clicking the button the effect node.

**Comment by age3rcm** [comment]
what about the anti-aliasing?

**Comment by age3rcm** [comment]
there is more to a scene than just disabling all effects. in a complex scene with multiple cut-out rigs and a lot of columns, to disable specific effects wich cause higher rendering times would be time consuming and suceptible to left-overs

**Comment by age3rcm** [comment]
another thing that if you simply disable everything on fx view, you end up disabling core columns too.
but disabling all effects on fx schematic all at once is not possible

**Comment by RodneyBaker** [comment]
You may wish to investigate OCA export as that will ignore FX which leads to a rather speedy rendering.
I believe this also automatically disregards antialiasing although we'd need to confirm that.

There is a difference between exporting and rendering to screen via OpenGL so we need to clarify that specific part a little more concerning your request.

What you could do is run multiple instances of Opentoonz, one of which points to the output of OCA images while the other (master) contains all the additional bells and whistles (such as fx) that would be included in a fuller final rendering.

Here's an example of OCA output (on the left) vs preview rendering with effects (on the right):
![image](https://user-images.githubusercontent.com/2576432/196056357-40e883f6-6626-43ad-be95-d88b7874209d.png)

**Comment by age3rcm** [comment]
i cannot find on opentoonz manual how to make an OCA export. But this is the desired result from this request!

**Comment by age3rcm** [comment]
the quick render would show exactly what you see on the left panel, what you see on the canvas

**Comment by RodneyBaker** [comment]
If don't have the OCA Export option under File > Export... you'll need to download a current nightly release to get the feature.

![image](https://user-images.githubusercontent.com/2576432/196196045-225ffe21-8da6-49ab-b18f-5d59839e77ba.png)

There are a few options that are useful at the bottom of the export dialogue; to allow exporting to SVG vector format (instead of converting the vectors to raster, and to export out to the EXR image format.

![image](https://user-images.githubusercontent.com/2576432/196196719-145a7161-3507-416d-8c71-5a2bf5a524b0.png)

**Comment by age3rcm** [comment]
good news! OCA export does have the feature i suggested BUT
why did it export every individual level?
![MLP-FIM_Mare_oca explorer_xZJxDQ8qdv](https://user-images.githubusercontent.com/36406840/196238557-c3eb6b19-6610-4e7f-b508-26399922e3b2.png)
![MLP-FIM_Mare_ MLP-FIM_Mare __OpenToonz_1 6 OpenToonz_bvm1F5vilV](https://user-images.githubusercontent.com/36406840/196238563-5aa4bd0a-3533-4aaa-b35b-5a75a1a20fe2.png)

**Comment by RodneyBaker** [comment]
> is there a way to have "vanilla" opentoonz and also Opentoonz nightly installed?

Sure.  Any release of Opentoonz can be made into a (Windows) portable and each can then be ran from it's own directory as needed... and even at the same time.

This is probably not the ideal place to run through the steps to create a portable as there are variations depending on how we approach it.

The core steps though are:
- Copy both the Opentoonz program directory and the stuff directory into a new directory (say... on your desktop)
- Rename the 'Opentoonz stuff' directory as 'portablestuff'
Done.  Run the program

**Comment by age3rcm** [comment]
for the OCA export, is there a way to just render the frames instead of individual pieces too?

**Comment by age3rcm** [comment]
i noticed that the vectors that comes along with it are only the exposure of level, rather than repeated frames

**Comment by RodneyBaker** [comment]
Yes, OCA stands for open 'cel' animation formation where the cel is not rendered again when as that would require needless rerendering of frames.  Instead the instruction of exposures is maintained in the .JSON text file so the target application knows to repeat that frame across the designated exposures.

I don't know of any way to have the export composite layers together as that also would defeat the purpose of cel animation layers placed in front of a camera.

Perhaps @justburner can be convinced to investigate if combining levels via some form of merge would be practical.
I would imagine that would consist of a checkbox that triggers a pre-render that combines all of the columns first prior to export but... that would largely be like doing this ourselves via a normal render.

**Comment by age3rcm** [comment]
it did export full frames alongside the separated layers. is it possible to avoid that and let it render the frames without effects only?

**Comment by RodneyBaker** [comment]
> it did export full frames alongside the separated layers. is it possible to avoid that and let it render the frames without effects only?

That is currently not an option in the initial release of the OCA export feature.
Perhaps for the next iteration of updates for OCA export that (and other useful options) will be considered.

cc:  @justburner @JPUser1
